app/models/usergroup_member.rb

```diff
@@ -1,4 +1,24 @@
 class UsergroupMember < ApplicationRecord
+  module MemberOverride
+    # the belongs to would apply default scope limiting the search by taxonomies but we need to delete
+    # all members, regardless of their taxonomies
+    #
+    # we can't use custom scope definition in belongs_to because it's polymorphic, unscope(:where) would reset
+    # the member_type condition so we need to override how we search for members
+    def member
+      case member_type
+        when 'User'
+          User.unscoped { super }
+        when 'Usergroup'
+          Usergroup.unscoped { super }
+        else
+          raise "Unknown member type #{member_type}"
```

Comment:
... `Foreman::Exception.new()` ?

Reply:
Good catch. We raise it as `ArgumentError` on two other locations in this file. Whatever exception type we use, we should be consistent.

Reply:
I agree, I changed the exception to ArgumentError since this is an internal exception for devs, not really for users.